### ISSUE TITLE
Opencache.uk authentication issue

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#6153]
##### Detailed steps causing the problem:
- oc.uk is not active
- Activate the connector in settings
- Select to start the procedure
- Open with browser and enter login data on OC page
- Select "Login" on oc.uk page

##### Actual behavior after performing these steps:
The system offers to either use "Browser" or "c:geo" after selecting "Login" on the authentication website
Selecting "Browser" tries to open `callback://...` in the browser with cannot work
Selecting "c:geo" shows the error "c:geo cannot open this cache. Is it really a geocache" (or similar)

##### Expected behavior after performing these steps:
Automatic return to c:geo after selecting OK.
This is working with other OC-sites (tested oc.nl on the same device)

##### Version of c:geo used:
2016.11.21-RC

##### Is the problem reproducible for you?
Yes

##### Other comments and remarks:
Not sure if this is related to my device or recent changes

**Comment by rsudev** [comment]
There is a slight issue WRT especially to opencache.uk. There is an overlap between the cache urls, handled by the cache-detail activity (http://opencache.uk/ok...) and the url for the oauth handshake (http://opencache.uk/okapi/...).
On my device (Android 6.0.1) I had to activate 'Always ask' for the app-handled urls, in order to be able to complete the authorization. But than it worked  with selecting browser in the first, and c:geo in the second step. Which Android version do you have?

**Comment by rsudev** [comment]
Re-checking, I wanted to mention that I use firefox as a browser and I was not asked for a selection on the second step.

**Comment by Lineflyer** [comment]
> There is an overlap between the cache urls, handled by the cache-detail activity (http://opencache.uk/ok...) and the url for the oauth handshake (http://opencache.uk/okapi/...).

How is this for opencaching.nl. Do we have the same overlap there?
Beacuse I tested oc.nl on the same device without a problem.

**Comment by rsudev** [comment]
No, because the okapi urls always start with ok, but only opencache.uk has cache codes starting with ok and therefore have this clash

**Comment by Lineflyer** [comment]
I think I have to use an activity interceptor to check what is triggering what here.

**Comment by Bananeweizen** [comment]
I remember that I also had to manually select the browser and cgeo during the oauth dance when testing the oc.uk changes.

**Comment by Lineflyer** [comment]
I had no time to analyse in depth but I recorded a video for you.
Device is a factory setting Samsung S4 with Cyanogenmod 13 (Android 6.0.1).

https://www.dropbox.com/s/yyqn805dc90kq5f/20161128_231004.mp4?dl=0

**Comment by Lineflyer** [comment]
My assumption (in my words):
It seems the ```callback``` is "catched" as a geocode URL instead of the callback URL or a new instance of c:geo is started instead of "coming back" to where I started which leads to this problem.

**Comment by Lineflyer** [comment]
There are users out there who are using this without a problem (just had a conversation with some). So it might be a device issue on my side.
As I did not have time to dig deeper, lets close this for now.

**Comment by SammysHP** [comment]
There are also users who have this problem.

https://twitter.com/Daddycooke/status/820364639003406336

**Comment by Lineflyer** [comment]
Workaround info, which sometimes worked for the affected users:
Reset association ("Open with") for c:geo and your default browser in the device appmanager. Start the authorization again and after starting the procedure select your browser "once" at the first prompt and select c:geo "once" at the next prompt.

**Comment by Lineflyer** [comment]
Still reports for this on support mail

**Comment by alopix** [comment]
Interesting one. I was able to sign in originally but cannot re-authenticate now. It shows exactly the error mention in the original description. Have not checked the workarounds yet.

I must be in a weird state with oc.uk in general, because I could view a cache in c:geo this weekend, and now that I am trying to log it, it complains that I am either offline or signed out (which is why I have found this bug still being an issue).
